File: java/WebFormsCore/Security.java
package WebFormsCore;

// WebForms.java 2.0 - The Back-End Part of WebForms Core Technology, Owned by Elanat (https://elanat.net)
// Compatible with WebFormsJS version 2.0

public class Security {
    public String safeValue(String value) {
        if (value == null || value.length() < 1)
            return value;

        if (value.charAt(0) == '@') {
            value = value.substring(1);
            value = "$[at];" + value;
        }

        value = value.replace("\n", "$[ln];");
        value = value.replace("|", "$[vb];");
        value = value.replace(",@", "$[co];@");

        return value;
    }
}

File: elixir/webforms_core/security.ex
# webforms.ex 2.1 - The Back-End Part of WebForms Core Technology, Owned by Elanat (https://elanat.net)
# Compatible with WebFormsJS version 2.1

defmodule WebFormsCore.Security do
  def safe_value(value) do
    if value == "" do
      value
    else
      value = if String.first(value) == "@", do: "@" <> value, else: value

      value
      |> String.replace("\n", "$[ln];")
      |> String.replace(",@", "$[co];@")
      |> String.replace(<<28>>, <<0>>)
      |> String.replace(<<29>>, <<0>>)
      |> String.replace(<<30>>, <<0>>)
      |> String.replace(<<31>>, <<0>>)
    end
  end
end